FreeRTOS/FreeRTOS-Plus-TCP
# Fix unit test build issues and test failures for IP

Description
-----------
This PR fixes unit test build issues and test failures for IP timers unit tests which were failing after the latest IPv6 integration changes. Creates 2 new diffconfig for IP - diffconfig1 and diffconfig2 for enabling/disabling `ipconfigCOMPATIBLE_WITH_SINGLE` for the currently existing IP diffconfigs.

Also fixes:
- Build issue with IP utils diffconfig
- Minor fix with an ARP unit test - `test_vARPRefreshCacheEntry_IPAndMACInDifferentLocations1`

Test Steps
-----------
```
cmake -S test/unit-test -B test/unit-test/build/ 
make -C test/unit-test/build/ all 
cd test/unit-test/build/
ctest -E system --output-on-failure --verbose

```
Related Issue
-----------
IP unit test cases were failing due to the changes in the source code as part of the IPv6 integration


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
